### PR TITLE
feature/dao-membership-profile-page (PRO-616)

### PR DESCRIPTION
**Pull request by jonathanprozzi**
## Linear Ticket

- [PRO-616](https://linear.app/govrn/issue/PRO-616/dao-card-on-profile-page)
- [PRO-779](https://linear.app/govrn/issue/PRO-779/states)
- [PRO-617](https://linear.app/govrn/issue/PRO-617/show-daos-youve-joined)
- [PRO-817](https://linear.app/govrn/issue/PRO-817/create-feature-flag-component)

## Description

- Adds in the UI for the _My DAOs_ section of the Profile
- Modifies our Select components to support better portalling and respect to clickable elements behind the open menu
- Adds the `ProfileDaos` and `DaoCard` components
- Fetching is done at the `ProfileDaos` component so that we don't block any loading for the rest of the Profile (similar to how we did this for the DAO Dashboards)
- Uses mock data that'll be replaced:
  - Noted throughout the code but the mock data is for the user's DAOs that (mock data being used for the `favorite` and their `role`)  

## Screencaptures

govrn-profile-daos-mobile
govrn-profile-daos

### CODE REVIEW

**Comment by alexkeating** [comment]
One UI thing I noticed, maybe a gap should be enforced
Screenshot from 2023-01-16 14-53-59

File: apps/protocol-frontend/src/components/DaoCard.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            ) : (
+              <IconButton
+                aria-label="Click on the star to favorite this DAO."
+                bg="transparent"
+                _hover={{ bg: 'transparent' }}
+                _active={{ bg: 'transparent' }}
+                icon={
+                  dao.favorite === true ? (
+                    <HiStar
+                      fill={dao.role === 'recruit' ? 'gray.700' : '#5100E4'}
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/DaoCard.tsx:120]: This turnary should probably be a constant to be dryer

File: apps/protocol-frontend/src/components/ProfileDaos.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+// only here for mocking purposes, will replace
+type DaoRoles = 'admin' | 'contributor' | 'recruit';
+
+type Dao = {
+  id: number;
+  name: string;
+  role: DaoRoles;
+  favorite: boolean;
+};
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ProfileDaos.tsx:16]: Some code duplication here, with dao card

**Review by alexkeating — COMMENTED** [review]
Looks good maybe some small nits

File: apps/protocol-frontend/src/components/DaoCard.tsx (hunk 1 above)
Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/DaoCard.tsx:120]: Good calls on all of these especially the gap -- will resolve.

**Comment by jonathanprozzi** [comment]
With gap more explicitly set:

dao-grid-gap

File: apps/protocol-frontend/src/components/ProfileDaos.tsx (hunk 2 above)
Comment (jonathanprozzi) [reply apps/protocol-frontend/src/components/ProfileDaos.tsx:16]: Oh yeah -- figure once we remove the mockData that'll use the same type in both places (likely our UIGuild type)

**Comment by amsos** [comment]
@jonathanprozzi looking good

On the dropdown list to the left of the `join` button, I click the carrot, it shows me daos I can join, then I click the DAO, click the join button, and then it adds me as a `recruit`.

Is that the right flow?

**Comment by jonathanprozzi** [comment]
> @jonathanprozzi looking good
> 
> 
> 
> On the dropdown list to the left of the `join` button, I click the carrot, it shows me daos I can join, then I click the DAO, click the join button, and then it adds me as a `recruit`.
> 
> 
> 
> Is that the right flow?

Thanks! That is the flow. We have a separate ticket for that integration but that is how it'll work.

**Pull request merged**